src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -60,58 +63,14 @@ public void onPost() {
      * Called when a POST is made.
      */
     public void onPost(String triggeredByUser) {
-        final String pushBy = triggeredByUser;
-        getDescriptor().queue.execute(new Runnable() {
-            private boolean runPolling() {
-                try {
-                    StreamTaskListener listener = new StreamTaskListener(getLogFile());
-
-                    try {
-                        PrintStream logger = listener.getLogger();
-                        long start = System.currentTimeMillis();
-                        logger.println("Started on "+ DateFormat.getDateTimeInstance().format(new Date()));
-                        boolean result = job.poll(listener).hasChanges();
-                        logger.println("Done. Took "+ Util.getTimeSpanString(System.currentTimeMillis()-start));
-                        if(result)
-                            logger.println("Changes found");
-                        else
-                            logger.println("No changes");
-                        return result;
-                    } catch (Error e) {
-                        e.printStackTrace(listener.error("Failed to record SCM polling"));
-                        LOGGER.log(Level.SEVERE,"Failed to record SCM polling",e);
-                        throw e;
-                    } catch (RuntimeException e) {
-                        e.printStackTrace(listener.error("Failed to record SCM polling"));
-                        LOGGER.log(Level.SEVERE,"Failed to record SCM polling",e);
-                        throw e;
-                    } finally {
-                        listener.close();
-                    }
-                } catch (IOException e) {
-                    LOGGER.log(Level.SEVERE,"Failed to record SCM polling",e);
-                }
-                return false;
-            }
+        throw new UnsupportedOperationException("deprecated");
+    }
 
-            public void run() {
-                if (runPolling()) {
-                    String name = " #"+job.getNextBuildNumber();
-                    GitHubPushCause cause;
-                    try {
-                        cause = new GitHubPushCause(getLogFile(), pushBy);
-                    } catch (IOException e) {
-                        LOGGER.log(Level.WARNING, "Failed to parse the polling log",e);
-                        cause = new GitHubPushCause(pushBy);
-                    }
-                    if (job.scheduleBuild(cause)) {
-                        LOGGER.info("SCM changes detected in "+ job.getName()+". Triggering "+name);
-                    } else {
-                        LOGGER.info("SCM changes detected in "+ job.getName()+". Job is already in the queue");
-                    }
-                }
-            }
-        });
+    public void onPost(String triggeredByUser, URIish repository, String sha1, String ref) {
+        String branch = ref.substring("refs/heads/".length());
```

Comment:
As i understand this can prevent from tracking any `refs/*` and not `refs/head` (that is just default of "--heads" and branches). Why do you need hardcode `refs/heads` at all?